tests/parametric/test_otel_span_methods.py

```diff
@@ -398,6 +400,204 @@ def test_otel_set_attributes_separately(self, test_agent, test_library):
+        assert link.get("trace_id") == 8687463697196027922
+        assert link.get("trace_id_high") == 1311768467284833366
+
+        assert link.get("tracestate") is not None
+        print("tracestate is '%s'" % link["tracestate"])
+        tracestateArr = link["tracestate"].split(",")
+        assert len(tracestateArr) == 3
+        dd_num = 0 if tracestateArr[0].startswith("dd=") else 1
+        other_num = 0 if dd_num == 1 else 1
+        assert tracestateArr[other_num] == "foo=1"
+        assert tracestateArr[2] == "bar=baz"
+        tracestateDD = tracestateArr[dd_num][3:].split(";")
+        assert len(tracestateDD) == 2
+        assert "s:2" in tracestateDD
+        assert "t.dm:-4" in tracestateDD
+
+        assert link.get("flags") == 1 | -2147483648  # Sampled and Set (31 bit according the RFC)
+        assert len(link.get("attributes")) == 0
+
+    @missing_feature(context.library < "java@1.26.0", reason="Implemented in 1.26.0")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    def test_otel_span_started_with_link_from_other_spans(self, test_agent, test_library):
+        """Test adding a span link from a span to another span.
+        """
+        with test_library:
+            with test_library.otel_start_span("root") as parent:
+                parent.end_span()
+                with test_library.otel_start_span("first", parent_id=parent.span_id) as first:
+                    first.end_span()
+                with test_library.otel_start_span(
+                    "second",
+                    parent_id=parent.span_id,
+                    links=[
+                        Link(parent_id=parent.span_id),
+                        Link(parent_id=first.span_id, attributes={"bools": [True, False], "nested": [1, 2]}),
+                    ],
+                ) as second:
+                    second.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="root"))
+        assert len(trace) == 3
+
+        root = find_span(trace, otel_span(name="root"))
+        root_tid = root["meta"].get("_dd.p.tid") or "0" if "meta" in root else "0"
+
+        first = find_span(trace, otel_span(name="first"))
+        second = find_span(trace, otel_span(name="second"))
+        assert second.get("parent_id") == root.get("span_id")
+
+        span_links = retrieve_span_links(second)
+        assert span_links is not None
+        assert len(span_links) == 2
+
+        link = span_links[0]
+        assert link.get("span_id") == root.get("span_id")
+        assert link.get("trace_id") == root.get("trace_id")
+        assert link.get("trace_id_high") == int(root_tid, 16)
+        assert len(link.get("attributes")) == 0
+        assert link.get("tracestate") == ""
```

Comment:
In python a span will always have at least two values in the tracestate (sampling decision and sampling priority). This tracestate value comes from the parent span's context ([python implementation](https://github.com/DataDog/dd-trace-py/blob/28b75850a72f670e5edc2bfa6c3935a7e2d30759/ddtrace/span.py#L546)). This is similar to the opentelemetry implementation. 

Should we adjust the the tracestate assertion to check the value if it is set? Should we make this behavior consistent across languages? 

```suggestion
        if link.get("tracestate"):
            assert link.get("tracestate") == "dd=s:1;t.dm:-0"
```

Reply:
> Should we adjust the the tracestate assertion to check the value if it is set? 

For now, I keep it consistent with the original test: https://github.com/DataDog/system-tests/pull/1811/files#diff-e63af22f5cd81bf600c30dfb393aef024e358dd336ede3786d147f929cab2b54R181
If we do any change, I think we should do it in a dedicated PR for both OTel and DD API.

> Should we make this behavior consistent across languages?

I don't think we need. It is costly to generate a `tracestate` and might not meet the performance requirements for the Java tracer.
We do need to state the benefit of such data before requiring to support it in the RFC.